server/middleware/retrieveTopicList.js

```diff
@@ -0,0 +1,14 @@
+module.exports = cmsService => async (req, res, next) => {
+  try {
+    const { establishmentName } = req.session;
+
+    if (!establishmentName) {
+      throw new Error('Could not determine establishment!');
+    }
+    res.locals.allTopics = await cmsService.getTopics(establishmentName);
+    next();
+  } catch (e) {
+    e.message = `Error loading topics: ${e.message}`;
+    next(e);
+  }
+};
```

Comment:
@jonhindle  we've altered some tests to cover it but not testing it anger - we'll need to think if we can both use the same middleware and merge this into yours or not.